UITests/StepDefinitions/ProductStepsAssertions.cs

```diff
@@ -0,0 +1,66 @@
+﻿using CatalogServiceAPI.Models.StepsModels;
+using NUnit.Framework;
+using TechTalk.SpecFlow;
+
+namespace UserManagementServiceUITests.StepDefinitions
+{
+    [Binding]
+    public class ProductStepsAssertions
+    {
+        private readonly DataContext _context;
+
+        public ProductStepsAssertions(DataContext context)
+        {
+            _context = context;
+        }
+
+        [StepDefinition(@"The product with a certain data is displayed on the page")]
+        public void ThenTheProductWithACertainDataIsDisplayedOnThePage(List<ProductModel> productsInfo)
+        {
+                _context.MainPage.WaitAmountOfExpectedProducts(productsInfo.Count);
+
+                var actualNames = _context.MainPage.GetProductNamesText();
+                var actualManufactors = _context.MainPage.GetProductManufactorsText();
+               
+                Assert.Multiple(() =>
+                {
+                    CollectionAssert.AreEquivalent(actualNames, productsInfo.Select(product => product.Name).ToList());
+                    CollectionAssert.AreEquivalent(actualManufactors, productsInfo.Select(product => product.Manufactor).ToList());
+                });
+        }
```

Comment:
If, before or during the tests, the database gets products, for example, with names that match the test data of your test, then the search will work correctly, but the test will fail because it checks the expected results with a list of actual ones, which may be more than expected. Maybe you should reconsider your assertions.

Reply:
To be discussed